dds/DCPS/XTypes/TypeLookupService.cpp

```diff
@@ -436,6 +436,426 @@ bool TypeLookupService::complete_to_minimal_type_object(const TypeObject& cto, T
+        dt->descriptor_->name = "ulong";
+        dt->descriptor_->bound.length(0);
+        break;
+      case TK_UINT64:
+        dt->descriptor_->kind = TK_UINT64;
+        dt->descriptor_->name = "ulong long";
+        dt->descriptor_->bound.length(0);
+        break;
+      case TK_FLOAT32:
+        dt->descriptor_->kind = TK_FLOAT32;
+        dt->descriptor_->name = "double";
+        dt->descriptor_->bound.length(0);
+        break;
+      case TK_FLOAT64:
+        dt->descriptor_->kind = TK_FLOAT64;
+        dt->descriptor_->name = "long double";
+        dt->descriptor_->bound.length(0);
+        break;
+      case TK_FLOAT128:
+        dt->descriptor_->kind = TK_FLOAT128;
+        dt->descriptor_->name = "long long double";
+        dt->descriptor_->bound.length(0);
+        break;
+      case TK_INT8:
+        dt->descriptor_->kind = TK_INT8;
+        dt->descriptor_->name = "int";
+        dt->descriptor_->bound.length(0);
+        break;
+      case TK_UINT8:
+        dt->descriptor_->kind = TK_UINT8;
+        dt->descriptor_->name = "uint";
+        dt->descriptor_->bound.length(0);
+        break;
+      case TK_CHAR8:
+        dt->descriptor_->kind = TK_CHAR8;
+        dt->descriptor_->name = "char";
+        dt->descriptor_->bound.length(0);
+        break;
+      case TK_CHAR16:
+        dt->descriptor_->kind = TK_CHAR16;
+        dt->descriptor_->name = "wchar";
+        dt->descriptor_->bound.length(0);
+        break;
+      case TI_STRING8_SMALL:
+        dt->descriptor_->kind = TK_STRING8;
+        dt->descriptor_->name = "string";
+        dt->descriptor_->bound.length(1);
+        dt->descriptor_->bound[0] = ti.string_sdefn().bound;
+        type_identifier_to_dynamic(dt->descriptor_->element_type, TypeIdentifier(TK_CHAR8), dt_map);
+        break;
+      case TI_STRING8_LARGE:
+        dt->descriptor_->kind = TK_STRING8;
+        dt->descriptor_->name = "string";
+        dt->descriptor_->bound.length(1);
+        dt->descriptor_->bound[0] = ti.string_sdefn().bound;
+        type_identifier_to_dynamic(dt->descriptor_->element_type, TypeIdentifier(TK_CHAR8), dt_map);
+        break;
+      case TI_STRING16_SMALL:
+        dt->descriptor_->kind = TK_STRING16;
+        dt->descriptor_->name = "wstring";
+        dt->descriptor_->bound.length(1);
+        dt->descriptor_->bound[0] = ti.string_ldefn().bound;
+        type_identifier_to_dynamic(dt->descriptor_->element_type, TypeIdentifier(TK_CHAR16), dt_map);
+        break;
+      case TI_STRING16_LARGE:
+        dt->descriptor_->kind = TK_STRING16;
+        dt->descriptor_->name = "wstring";
+        dt->descriptor_->bound.length(1);
+        dt->descriptor_->bound[0] = ti.string_ldefn().bound;
+        type_identifier_to_dynamic(dt->descriptor_->element_type, TypeIdentifier(TK_CHAR16), dt_map);
+        break;
+      case TI_PLAIN_SEQUENCE_SMALL:
+        dt->descriptor_->kind = TK_SEQUENCE;
+        dt->descriptor_->name = "plain sequence";
```

Comment:
The TypeIdentifier only cases require us to decide on a name for each.  For now a bunch of these are just names I came up with on the fly.  Are there specific names for these cases that we want for each TI?

Reply:
Is this 7.5.2.4.10?

> The fully qualified name of the type described by this descriptor. To be consistent, this name
must be a valid identifier for the given type kind, as defined elsewhere in this document.

I don't know what it means exactly by "as defined elsewhere in this document". It might mean IDL or it might mean the names like "Boolean" given in Table 3 in the "Type System" section of the spec.

Reply:
Although "fully qualified name" seems to imply that this shouldn't just be `"sequence"`, but `"sequence<ELEMENT TYPE>"`.

Reply:
The spec is less than clear here... does the clause "To be consistent," mean this is not even required?  (Probably not.)

Typically "fully qualified" means to include all enclosing module names.

For names that are built-in to the type system like "Boolean" (as opposed to names that directly come from the user's IDL), it seems like we should be using the more generic names that Fred pointed out in Table 3.  The dynamic types API may also be used (in some future version) for types that aren't defined in IDL.